src/foam/flow/Document.js

```diff
@@ -379,6 +383,58 @@ foam.CLASS({
           };
         },
 
+        'foam-with-contents': function(v) {
```

Comment:
Okay for now, but have a look at how foam1 let's you specify property values with sub-tags.